http/src/test/scala/org/broadinstitute/dsde/workbench/leonardo/dao/sam/SamServiceInterpSpec.scala

```diff
@@ -4,30 +4,32 @@ import akka.http.scaladsl.model.StatusCodes
 import cats.effect.IO
 import cats.effect.unsafe.implicits.global
 import org.broadinstitute.dsde.workbench.client.sam.ApiException
-import org.broadinstitute.dsde.workbench.client.sam.api.{AzureApi, GoogleApi, ResourcesApi}
-import org.broadinstitute.dsde.workbench.client.sam.model.{FullyQualifiedResourceId, GetOrCreateManagedIdentityRequest}
+import org.broadinstitute.dsde.workbench.client.sam.api.{AzureApi, GoogleApi, ResourcesApi, UsersApi}
+import org.broadinstitute.dsde.workbench.client.sam.model._
 import org.broadinstitute.dsde.workbench.leonardo.CommonTestData._
 import org.broadinstitute.dsde.workbench.leonardo.TestUtils.appContext
 import org.broadinstitute.dsde.workbench.leonardo.auth.CloudAuthTokenProvider
-import org.broadinstitute.dsde.workbench.leonardo.{LeonardoTestSuite, SamResourceType}
+import org.broadinstitute.dsde.workbench.leonardo.model.{ForbiddenError, LeoInternalServerError}
+import org.broadinstitute.dsde.workbench.leonardo.{LeonardoTestSuite, RuntimeAction, SamResourceType}
 import org.http4s.headers.Authorization
 import org.http4s.{AuthScheme, Credentials}
-import org.mockito.ArgumentMatchers
-import org.mockito.ArgumentMatchers.{any, anyList}
-import org.mockito.Mockito.when
+import org.mockito.ArgumentMatchers.{any, anyList, anyString}
+import org.mockito.Mockito.{doNothing, doThrow, verify, when}
+import org.mockito.{ArgumentCaptor, ArgumentMatchers}
 import org.scalatest.BeforeAndAfterAll
 import org.scalatest.funspec.AnyFunSpecLike
 import org.scalatestplus.mockito.MockitoSugar
 
 import java.util.UUID
+import scala.jdk.CollectionConverters._
 
 class SamServiceInterpSpec extends AnyFunSpecLike with LeonardoTestSuite with BeforeAndAfterAll with MockitoSugar {
```

Comment:
Lot of tests here but I think it has pretty complete coverage of `SamServiceInterp`. It completely uses mocks so runs very fast. I tried to organize test cases a bit with ScalaTest [FunSpec](https://www.scalatest.org/getting_started_with_fun_spec).